reactor-core/src/test/java/reactor/core/publisher/SinkManyBestEffortTest.java

```diff
@@ -0,0 +1,369 @@
+import static org.assertj.core.api.Assertions.fail;
+
+class SinkManyBestEffortTest {
+
+	@Test
+	void currentContextReflectSubscriberContext() {
+		SinkManyBestEffort<Integer> sink = SinkManyBestEffort.createBestEffort();
+
+		assertThat(sink.currentContext()).isSameAs(Context.empty());
+
+		AssertSubscriber<Integer> subscriber1 = new AssertSubscriber<>(Context.of("key", "value1"));
+		AssertSubscriber<Integer> subscriber2 = new AssertSubscriber<>(Context.of("key", "value2"));
+		sink.subscribe(subscriber1);
+		sink.subscribe(subscriber2);
+
+		assertThat(sink.currentContext()).isEqualTo(subscriber1.currentContext());
+	}
+
+	@Test
+	void tryEmitNextNoSubscribers() {
+		SinkManyBestEffort<Integer> sink = SinkManyBestEffort.createBestEffort();
+
+		assertThat(sink.tryEmitNext(1)).as("tryEmitNext").isEqualTo(Emission.FAIL_ZERO_SUBSCRIBER);
+	}
+
+	@Test
+	void tryEmitNextOnTerminated() {
+		SinkManyBestEffort<Integer> sink = SinkManyBestEffort.createBestEffort();
+		sink.tryEmitComplete().orThrow();
+
+		assertThat(sink.tryEmitNext(1)).as("tryEmitNext").isEqualTo(Emission.FAIL_TERMINATED);
+	}
+
+	@Test
+	void tryEmitCompleteNoSubscribersRetains() {
+		SinkManyBestEffort<Integer> sink = SinkManyBestEffort.createBestEffort();
+
+		assertThat(sink.tryEmitComplete()).as("tryEmitComplete").isEqualTo(Emission.OK);
+
+		StepVerifier.create(sink.asFlux())
+		            .verifyComplete();
+	}
+
+	@Test
+	void tryEmitCompleteOnTerminated() {
+		SinkManyBestEffort<Integer> sink = SinkManyBestEffort.createBestEffort();
+		sink.tryEmitComplete().orThrow();
+
+		assertThat(sink.tryEmitComplete()).as("tryEmitComplete").isEqualTo(Emission.FAIL_TERMINATED);
+	}
+
+	@Test
+	void tryEmitErrorNoSubscribersRetains() {
+		SinkManyBestEffort<Integer> sink = SinkManyBestEffort.createBestEffort();
+
+		assertThat(sink.tryEmitError(new IllegalStateException("boom"))).as("tryEmitError").isEqualTo(Emission.OK);
+
+		StepVerifier.create(sink.asFlux())
+		            .verifyErrorMessage("boom");
+	}
+
+	@Test
+	void tryEmitErrorOnTerminated() {
+		SinkManyBestEffort<Integer> sink = SinkManyBestEffort.createBestEffort();
+		sink.tryEmitError(new IllegalStateException("boom")).orThrow();
+
+		assertThat(sink.tryEmitError(new IllegalStateException("boom ignored"))).as("tryEmitError").isEqualTo(Emission.FAIL_TERMINATED);
+
+		StepVerifier.create(sink.asFlux())
+		            .verifyErrorMessage("boom");
+	}
+
+	@Test
+	void tryEmitErrorOnSubscribers() {
+		SinkManyBestEffort<Integer> sink = SinkManyBestEffort.createBestEffort();
+
+		AssertSubscriber<Integer> sub1 = AssertSubscriber.create();
+		AssertSubscriber<Integer> sub2 = AssertSubscriber.create(0);
+		sink.subscribe(sub1);
+		sink.subscribe(sub2);
+
+		assertThat(sink.tryEmitError(new IllegalStateException("boom"))).as("tryEmitError").isEqualTo(Emission.OK);
+
+		sub1.assertErrorMessage("boom");
+		sub2.assertErrorMessage("boom");
+	}
+
+	@Test
+	void addSubscriberThatCancelsOnSubscription() {
```

Comment:
this test does not seem to assert anything

Reply:
fixed

Reply:
resolved ?